sdk/src/ingredient.rs

```diff
@@ -649,17 +660,45 @@ impl std::fmt::Display for Ingredient {
     }
 }
 
-#[derive(Default)]
-/// This defines optional actions when creating [`Ingredient`]s from files.
-pub struct IngredientOptions {
-    /// This allows setting the title for the ingredient. (If `None`, then the default behavior is to use the file's name.)
-    pub title: Option<&'static str>,
+/// This defines optional operations when creating [`Ingredient`] structs from files.
+#[cfg(feature = "file_io")]
+pub trait IngredientOptions {
```

Comment:
Disregard my previous comment (which I've now deleted). Allowing for overriding of hash and thumbnail generation  makes sense and justifies the change from struct to trait.

Reply:
updated